### PR TITLE
fix: use docker healthcheck

### LINKED ISSUE #8474: Collabora is not available
Step: 
- start ocis with wopi using https://github.com/owncloud/ocis/blob/master/deployments/examples/ocis_wopi/docker-compose.yml
or https://github.com/owncloud/web/blob/master/docker-compose.yml
- check if collabora is working

Actual: 
Screenshot 2024-02-16 at 10 16 50

image

app-provider-collabora log: 
```
2024-02-15 21:44:08 {"level":"fatal","service":"app-provider-collabora","error":"Failed to connect to NATS Server: nats: no servers available for connection","time":"2024-02-15T20:44:08Z","message":"Registration error for external service com.owncloud.api.app-provider-collabora"}
2024-02-15 21:44:09 {"level":"error","service":"app-provider-collabora","error":"unable to register services: rgrpc: grpc service appprovider could not be started,: Application server at https://collabora.owncloud.test does not match this AppProvider for Collabora","time":"2024-02-15T20:44:09Z","message":"error starting the grpc server"}
```

cc @2403905

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Description
Trying to fix #8474 without docker user 0

## How Has This Been Tested?
```shell
docker compose up -d
```

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: deployments/examples/ocis_wopi/docker-compose.yml [hunk 1]
```diff
@@ -133,22 +129,20 @@ services:
...
       - ocis-config:/etc/ocis
     logging:
       driver: ${LOG_DRIVER:-local}
     restart: always
     depends_on:
-      - ocis
+      ocis:
+        condition: service_started
+      collabora:
+        condition: service_healthy
```
Comment (wkloucek) [inline deployments/examples/ocis_wopi/docker-compose.yml:140]: https://github.com/owncloud/ocis/pull/8144#discussion_r1444196372

> This is only part of V2, right? V3 https://docs.docker.com/compose/compose-file/compose-file-v3/#depends_on doesn't mention it.

Comment (wkloucek) [reply deployments/examples/ocis_wopi/docker-compose.yml:140]: Ahh, actually the GO version of docker-compose seems to have it!??? https://docs.docker.com/compose/compose-file/05-services/#long-syntax-1

Comment (DeepDiver1975) [reply deployments/examples/ocis_wopi/docker-compose.yml:140]: local tests show this is working .... :shrug:

Comment (DeepDiver1975) [reply deployments/examples/ocis_wopi/docker-compose.yml:140]: running docker  26.1.0

**Review by micbar — APPROVED** [review]

**Pull request merged**